Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -400,24 +404,21 @@ extension HTTPClient {
     /// Response execution context. Will be created by the library and could be used for obtaining
     /// `EventLoopFuture<Response>` of the execution or cancellation of the execution.
     public final class Task<Response> {
-        /// `EventLoop` used to execute and process this request.
+        @available(*, deprecated, renamed: "eventLoop")
         public var currentEventLoop: EventLoop {
-            return self.lock.withLock {
-                _currentEventLoop
-            }
+            return self.eventLoop
         }
 
-        /// The stored property used by `currentEventLoop` in combination with the `lock`
-        ///
-        /// In most cases you should use `currentEventLoop` instead
-        private var _currentEventLoop: EventLoop
+        /// The `EventLoop` the delegate will be executed on.
```

Comment:
Maybe there should also be a `public` accessible property to the `channel`'s `EventLoop`?

Reply:
What would the use-case be?

Reply:
Happy to add now or later if we feel there’s a compelling-enough reason

Reply:
I was thinking about the backpressure futures of the delegate: performance wise, it's best to return one from the same event loop as the channel, which wouldn't be possible anymore if there is no access to the channel event loop.

Reply:
I'm inclined to say that if that really matters much, the user should've chosen `.delegateAndChannel(on: ...)` and therefore the channel's EL == delegate's EL. @Lukasa what do you think?

Reply:
The issue I see is that `.delegateAndChannel(on: ...)` is really restrictive for the connection pool and some delegates that don't need to do back-pressure (eg: `ResponseAccumulator`) would need to use it to avoid performance loss even if the `delegate`/`channel` EL doesn't matter for them.

Reply:
I think it’s reasonable to be able to ask for the delegate loop.

Reply:
@Lukasa The delegate `EventLoop` should already be accessible with the `eventLoop`, but I was thinking about the channel `EventLoop`.

To give a more concrete example of what worries me, if you look at what happens in the `ResponseAccumulator`, the newest changes are doing this:
```swift
public func didReceiveHead(task: HTTPClient.Task<Response>, _: HTTPResponseHead) -> EventLoopFuture<Void> {
    return task.eventLoop.makeSucceededFuture(())
}
```

But this means that in this part:
```
self.delegate.didReceiveBodyPart(task: self.task, body)
                    .hop(to: context.eventLoop)
```
The cost of hopping will be quite big especially something that's going to be called so much often.

Reply:
On the other hand if letting public access to the channel event loop risks introducing too many errors on lib user side, maybe we could actually allow returning `nil` for the delegates that don't need backpressure?

Reply:
Yeah, I don't think it's worth exposing the channel event loop at this time. I understand the concern, and I think we can address it in the future, but for now I don't think I'd really bother. The performance concern is not really large enough (IMO) to justify it right now.

Reply:
@adtrevor if you really need that kind of performance, I'd say just use `.delegateAndChannel(on: ...)` and then the delegate and the Channel's EL are guaranteed to be the same. If we truly find that there are important use-cases to get the Channel's El we can, as Cory points out, add that at any point without breaking API.